### PR TITLE
Diagnostics++

### PR DESCRIPTION
**Pull request by Dekken**
WIP placeholder PR for Teamcity builds

### CODE REVIEW

**Comment by Dekken** [comment]
The structure is in place, not everything is being written to the hdf5 file, but it should be easy to change to do so

File: src/amr/diagnostic/detail/samrai_highfive.h [hunk 1]
```diff
@@ -0,0 +1,379 @@
...
+}
+
+/*TODO
+  add time/iterations
+*/
+template<typename Model>
+void SamraiHighFiveDiagnostic<Model>::write(PatchLevel& level, std::string&& path,
+                                            Diagnostics const& diagnostics)
+{
+    size_t patch_idx = 0;
```
Comment (nicolasaunai) [inline src/amr/diagnostic/detail/samrai_highfive.h]: this id should be unique in parallel

File: tests/initializer/test_initializer.cpp [hunk 2]
```diff
@@ -1,28 +1,26 @@
...
-#include "data/grid/gridlayout.h"
-#include "data/grid/gridlayoutimplyee.h"
-#include "data/particles/particle_array.h"
-
-#include <memory>
+#include "diagnostic_manager.h"
+namespace PHARE
+{
+// global extern for python addition of diagnostics
+ADiagnosticsManager* diagnosticManager = new NoOpDiagnosticManager{};
```
Comment (nicolasaunai) [inline tests/initializer/test_initializer.cpp]: I don't understand

File: src/diagnostic/diagnostic_manager.h [hunk 3]
```diff
@@ -0,0 +1,263 @@
...
+void DiagnosticsManager<Writer>::dump(/*time iteration*/)
+{
+    size_t iter = 1; // TODO replace with time/iteration idx
+    auto active = [](size_t current, size_t start, size_t end, size_t every) {
+        return (current >= start && current <= end) && current % every == 0;
+    };
+    DiagnosticWritingList diagnosticWriters;
+    for (auto& diag : diagnostics)
+    {
+        if (active(iter, diag.start_iteration, diag.end_iteration, diag.compute_every))
```
[the remark this answers is not in the document]
Comment (Dekken) [reply src/diagnostic/diagnostic_manager.h]: I have merged your branch updates

File: CMakeLists.txt [hunk 4]
```diff
@@ -161,6 +181,9 @@ if (test)
           )
   ENDIF(coverage)
 
+  include_directories(
```
[the remark this answers is not in the document]
Comment (Dekken) [reply CMakeLists.txt]: to minimise duplication of code between tests.
hierarchy/etc

[the remark this answers is not in the document]
Comment (Dekken) [reply CMakeLists.txt]: to be replaced with main default headers

**Comment by Dekken** [comment]
replaced by https://github.com/PHAREHUB/PHARE/pull/69

**Pull request closed without merging**